src/app/screens/backupWalletSteps/verifySeed.tsx

```diff
@@ -32,54 +26,125 @@ const TransparentButtonContainer = styled.div((props) => ({
   width: '100%',
 }));
 
-const ButtonContainer = styled.div((props) => ({
-  marginLeft: props.theme.spacing(2),
-  width: '100%',
+const Heading = styled.h3((props) => ({
+  ...props.theme.body_l,
+  color: props.theme.colors.white[200],
+  marginBottom: props.theme.spacing(16),
 }));
 
-interface VerifySeedProps {
-  onVerifySuccess: () => void;
+const WordGrid = styled.div`
+  display: grid;
+  grid-template-columns: repeat(3, 1fr);
+  gap: ${(props) => props.theme.spacing(5)}px;
+  margin-bottom: ${(props) => props.theme.spacing(12)}px;
+`;
+
+const WordButton = styled.button`
+  ${(props) => props.theme.body_medium_m};
+  color: ${(props) => props.theme.colors.white['0']};
+  background-color: ${(props) => props.theme.colors.background.elevation3};
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  padding: ${(props) => props.theme.spacing(6)}px;
+  border-radius: ${(props) => props.theme.radius(1)}px;
+  text-transform: capitalize;
+  transition: all 0.1s ease;
+  :hover:enabled {
+    opacity: 0.8;
+  }
+  :active:enabled {
+    opacity: 0.6;
+  }
+`;
+
+const NthSpan = styled.span`
+  ${(props) => props.theme.body_bold_l};
+  color: ${(props) => props.theme.colors.white['0']};
+`;
+
+const ErrorMessage = styled.p<{ visible: boolean }>`
+  ${(props) => props.theme.body_s};
+  color: ${(props) => props.theme.colors.feedback.error};
+  visibility: ${(props) => (props.visible ? 'initial' : 'hidden')};
+`;
+
+const getOrdinal = (num: number): string => {
+  switch (num) {
+    case 1:
+      return '1st';
+    case 2:
+      return '2nd';
+    case 3:
+      return '3rd';
+    default:
+      return `${num}th`;
+  }
+};
+
+export default function VerifySeed({
+  onBack,
+  onVerifySuccess,
+  seedPhrase,
+}: {
   onBack: () => void;
+  onVerifySuccess: () => void;
   seedPhrase: string;
-}
-
-export default function VerifySeed(props: VerifySeedProps): JSX.Element {
-  const [seedInput, setSeedInput] = useState<string>('');
-  const [err, setErr] = useState('');
+}) {
   const { t } = useTranslation('translation', { keyPrefix: 'BACKUP_WALLET_SCREEN' });
-  const { onBack, onVerifySuccess, seedPhrase } = props;
+  const [err, setErr] = useState('');
+  const [correctCounter, setCorrectCounter] = useState(0);
+  const [quiz, setQuiz] = useState<{ words: string[]; answer: string; nth: string }>({
+    words: [],
+    answer: '',
+    nth: '',
+  });
 
-  const cleanMnemonic = (rawSeed: string): string =>
-    rawSeed.replace(/\s\s+/g, ' ').replace(/\n/g, ' ').trim();
+  const generateWords = useCallback(() => {
```

Comment:
Would this have a chance to generate a duplicate correct word which would show as incorrect if clicked on.

Reply:
Yeah that's a good one. Will handle that case

Reply:
handled in latest commit